### ISSUE TITLE
[Bug] Comptime IDE crash

### ISSUE THREAD

**Issue by aharabada** [beefytech/Beef#966]
The IDE crashes when I try to build the attached project.

[CompTimeTest.zip](https://github.com/beefytech/Beef/files/6213141/CompTimeTest.zip)

**Comment by aharabada** [comment]
I checked back at this issue and it seems the crash was fixed at some point.

However in the project I attached comptime still has some issues:
[Project.zip](https://github.com/beefytech/Beef/files/7491033/GlitchyEngine_Slim.zip)
The relevant classes are `GlitchyEngine.Math.Vector2` (also Vector3/4) and `GlitchyEngine.Math.SwizzleVectorAttribute`

- Compiling will throw two kinds of errors:
    - `Unable to comptime GlitchyEngine.Math.SwizzleVectorAttribute.ApplyToType(System.Type type)` for every struct with the SwizzleVectorAttribute (i.e. Vector2, Vector3 and Vector4)
    - `Type could not be found` for many (but not all) properties generated by the SwizzleVectorAttribute
- When changing a `const` field of one of the Vectors to `static readonly` the following error will appear `OnCompile const evaluation creates a data dependency during TypeInit during const-eval generation of 'System.Int.ToString(System.String outString, System.String format, System.IFormatProvider formatProvider)'`

It's also possible to tickle some crashes out of the IDE (though they are not always 100% reproducible):
- When changing a `const` field of one of the Vectors to `readonly` (not static) the IDE will crash
- Changes in SwizzleVectorAttribute.bf will often result in crashes (especially when the project was unsuccessfully built before). For example:
    - Removing `{i + 1}` in Line 97 and then saving will often crash the IDE
    - Making the size of arrays too small (e.g.: changing `positionState` (line 34) from `alphabet.Count` to `2` or `alphabet.Count - 1`, same goes for `posCount` though it crashes less often)
    - Changing `alphabet.Count` e.g. to `alphabet.Count + 1` in `for(int i = 0; i < alphabet.Count; i++)` (line 64) crashes the IDE sometimes

**Comment by bfiete** [comment]
That project triggered a number of good bugs.
Fixed at c5e2b78199b625ec4518031786012b1d70a35d6b

I believe all of the SwizzleVectorAttribute issues were the same. Let me know if any of these problems are still not resolved.

**Comment by aharabada** [comment]
The Project seems to be working good now. But I have bad (or good?) news: I found a different implementation of SwizzleVectorAttribute that still crashes the IDE:
```beef
using System;

namespace GlitchyEngine.Math
{
	/**
	 * Generates component swizzle operators for a vector.
	 */
	[AttributeUsage(.Struct | .Class)]
	struct SwizzleVectorAttribute : Attribute, IComptimeTypeApply
	{
		private int _vectorSize;

		public int VectorSize => _vectorSize;

		private String _vectorTypeName;

		this(int vectorSize, String vectorTypeName)
		{
			_vectorSize = vectorSize;
			_vectorTypeName = vectorTypeName;
		}

		/**
		 * Determines whether or not a swizzle operator can have a valid setter (e.g. no component assigned twice)
		*/
		static bool invalidSetter(int[4] cmp, int vecSize)
		{
			return cmp[0] == cmp[1] || (vecSize >= 3 && cmp[0] == cmp[2]) || (vecSize == 4 && cmp[0] == cmp[3]) ||
				(vecSize >= 3 && cmp[1] == cmp[2]) || (vecSize == 4 && cmp[1] == cmp[3]) ||
				(vecSize == 4 && cmp[2] == cmp[3]);
		}

		[Comptime]
		public void ApplyToType(Type type)
		{
			String[] componentNames = scope String[]("X", "Y", "Z", "W");

			for(int swizzleCount = 2; swizzleCount <= 4; swizzleCount++)
			{
				int[4] cmp = .();

				int cmp2max = swizzleCount > 2 ? _vectorSize : 1;
				int cmp3max = swizzleCount > 3 ? _vectorSize : 1;

				for(cmp[0] = 0; cmp[0] < _vectorSize; cmp[0]++)
				for(cmp[1] = 0; cmp[1] < _vectorSize; cmp[1]++)
				for(cmp[2] = 0; cmp[2] < cmp2max; cmp[2]++)
				for(cmp[3] = 0; cmp[3] < cmp3max; cmp[3]++)
				{
					String swizzleName = scope String(swizzleCount);
					String swizzleConstructor = scope String(swizzleCount * 3);
					String setter = scope String(128);

					bool setterInvalid = invalidSetter(cmp, swizzleCount);

					for(int c = 0; c < swizzleCount; c++)
					{
						swizzleName.Append(componentNames[cmp[c]]);

						if(c != 0)
						{
							swizzleConstructor.Append(", ");
						}
						swizzleConstructor.Append(componentNames[cmp[c]]);

						if(!setterInvalid && c < _vectorSize)
						{
							setter.AppendF($"\n\t\t{componentNames[cmp[c]]} = value.{componentNames[c]};");
						}
					}

					String swizzleString = scope $"""
						public {_vectorTypeName}{swizzleCount} {swizzleName}
						{{
							get => .({swizzleConstructor});
							set mut
							{{{setter}
							}}
						}}

						""";

					Compiler.EmitTypeBody(type, swizzleString);
				}
			}
		}
	}
}
```

**Comment by bfiete** [comment]
That code revealed some good comptime heap issues!
Fixed at ec61583db88f20af682d1b5016f3876f5c9fe945
